src/Core/Users/Actions/CreateUser.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace GetCandy\Api\Core\Users\Actions;
+
+use GetCandy;
+use GetCandy\Api\Core\Customers\Actions\FetchDefaultCustomerGroup;
+use GetCandy\Api\Core\Customers\Models\Customer;
+use GetCandy\Api\Core\Customers\Models\CustomerGroup;
+use GetCandy\Api\Core\Foundation\Actions\DecodeIds;
+use GetCandy\Api\Core\Languages\Actions\FetchDefaultLanguage;
+use GetCandy\Api\Core\Languages\Actions\FetchEnabledLanguageByCode;
+use GetCandy\Api\Core\Users\Resources\UserResource;
+use Lorisleiva\Actions\Action;
+
+class CreateUser extends Action
+{
+    /**
+     * Determine if the user is authorized to make this action.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the action.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'firstname' => 'required|string',
+            'lastname' => 'required|string',
+            'email' => 'required|email',
+            'password' => 'required|string|min:6',
+            'language' => 'nullable|string',
+            'details' => 'nullable|array',
+            'customer_groups' => 'nullable|array',
+        ];
+    }
+
+    /**
+     * Execute the action and return a result.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        $userModel = config('auth.providers.users.model', User::class);
+
+        $user = (new $userModel);
+        $user->name = $this->firstname.' '.$this->lastname;
+        $user->email = $this->email;
+        $user->password = bcrypt($this->password);
+
+        if (! $this->language) {
+            $user->language()->associate(FetchDefaultLanguage::run());
+        } else {
+            $user->language()->associate(FetchEnabledLanguageByCode::run([
+                'code' => $this->language,
+            ]));
+        }
+
+        $user->save();
+
+        $details = $this->details;
+        $details['firstname'] = $this->firstname;
+        $details['lastname'] = $this->lastname;
+        $details['fields'] = $this->details['fields'] ?? [];
+
+        $customer = Customer::create($details);
+        $customer->users()->save($user);
+
+        if ($this->customer_groups) {
+            $customer->customerGroups()->sync(
```

Comment:
Do we need to worry about this being in a separate action? i.e. would this duplicate on update?

Reply:
I did initially push this to an action you had setup but there were numerous issues, it's part of what I noted in discord and could do with a discussion around how we handle it

Reply:
As there is now the endpoints for customers what's the flow for a customer looking how things go I can remove customer creation from here and UpdateUser.php?

Do we also still want it to handle the customer groups if supplied as within the current context that's userland and not admin land only so it would just go to the default anyway?

Reply:
Yeah I think the issue is that a user can belong to a customer and a customer has many users. So if we're creating a customer when we create a user, we're unable to just create a user and assign them to an existing customer.

Customers and users should be separate, I guess if we pass a `customer_id` we assign them to that? 

Also with addresses, should these instead be attached to customer records as opposed to users? if so would we want to create a default customer for a user so everything stays in sync.

Reply:
With how it's set up now a user seems purely about authorising and less about information tied to them so it would make sense if the addresses were attached to the customer records.

I assume with orders that would end up being customer_id instead of user_id as it currently is too?

I guess the main thing is how it's handled because is it an issue if any user can become part of a customer if they know the encoded_id or can only an admin and those assigned to the customer add new people to that customer?
How did you envisage the current flow of sign up and a user being attached to a customer?